core/optimization/src/test/java/it/unibz/inf/ontop/iq/optimizer/UniqueConstraintInferenceTest.java

```diff
@@ -165,7 +165,7 @@ public void testUnion2() {
         IQTree tree = IQ_FACTORY.createNaryIQTree(
                 IQ_FACTORY.createUnionNode(ImmutableSet.of(X, A, B)),
                 ImmutableList.of(child1, child2));
-        assertEquals(ImmutableSet.of(ImmutableSet.of(X)), tree.inferUniqueConstraints());
+        assertEquals(ImmutableSet.of(ImmutableSet.of(X), ImmutableSet.of(X, A)), tree.inferUniqueConstraints());
```

Comment:
@DRovara , actually we should not add the second unique constraint, as it is already implied by the first one.
Can you please have a look?

The same remark holds for the following tests